### PR TITLE
DRY up finding the spec file

### PR DESCRIPTION
**Pull request by komidore64**

### CODE REVIEW

File: obal/data/roles/spec_file/meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+dependencies:
```
Comment (ekohl) [inline obal/data/roles/spec_file/meta/main.yml]: I think this also depends on `package_variables` to define `package_base_dir`.

Comment (komidore64) [reply obal/data/roles/spec_file/meta/main.yml]: `ensure_package` depends on `package_variables` so I didn't feel the need to specify it here also.

File: obal/data/roles/setup_sources/tasks/specfile.yml [hunk 2]
```diff
@@ -1,16 +1,10 @@
...
-- name: get spec file
-  find:
-    pattern: "*.spec"
-    path: "{{ package_dir }}"
-  register: setup_sources_spec_file
-
 - name: 'Set spec file path'
   set_fact:
-    setup_sources_spec_file_path: "{{ setup_sources_spec_file.files[0].path }}"
+    spec_file_path: "{{ spec_file.files[0].path }}"
```
Comment (ekohl) [inline obal/data/roles/setup_sources/tasks/specfile.yml]: This is only used in one place. Perhaps just use `spec_file.files[0].path` in the `spectool` command below?

File: obal/data/roles/diff_package/tasks/git_package_info.yml [hunk 3]
```diff
@@ -1,10 +1,4 @@
 ---
-- name: get spec file
-  find:
-    pattern: "*.spec"
-    path: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}"
-  register: spec_file
-
 - name: "Set spec file path, scl options and releasers"
```
Comment (ekohl) [inline obal/data/roles/diff_package/tasks/git_package_info.yml]: This looks outdated as well.

Comment (komidore64) [reply obal/data/roles/diff_package/tasks/git_package_info.yml]: Oh it sure is!

**Comment by komidore64** [comment]
> Given the repetition you may also consider setting `spec_file_path` in the `spec_file` role.

This is a good idea. I decided to set `spec_file_path` in the `spec_file` role.

File: obal/data/roles/diff_package/tasks/git_package_info.yml (hunk 3 above)
Comment (komidore64) [reply obal/data/roles/diff_package/tasks/git_package_info.yml]: fixed

File: obal/data/roles/setup_sources/tasks/specfile.yml (hunk 2 above)
Comment (komidore64) [reply obal/data/roles/setup_sources/tasks/specfile.yml]: I decided to move `spec_file_path` assignment into the `spec_file` role.

**Comment by evgeni** [comment]
> Thought experiment: should we become opinionated and prefer the spec filename to match the directory name (as the Fedora packaging guidelines suggest)? We now have an easy place to start enforcing it.

I'd prefer not to do this (yet), as that would require changes to all our downstream packaging repos.

**Comment by komidore64** [comment]
> I'd prefer not to do this (yet), as that would require changes to all our downstream packaging repos.

You beat me to saying this!

File: obal/data/playbooks/changelog/changelog.yaml [hunk 4]
```diff
@@ -4,14 +4,8 @@
...
   tasks:
-    - name: get spec file
-      find:
-        pattern: "*.spec"
-        path: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}"
-      register: spec_file
-
     - name: 'ensure changelog'
       changelog:
         spec: "{{ spec_file.files[0].path }}"
```
Comment (evgeni) [inline obal/data/playbooks/changelog/changelog.yaml]:
```suggestion
        spec: "{{ spec_file_path }}"
```

File: obal/data/playbooks/bump-release/bump-release.yaml [hunk 5]
```diff
@@ -6,14 +6,8 @@
...
+    - spec_file
   tasks:
-    - name: get spec file
-      find:
-        pattern: "*.spec"
-        path: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}"
-      register: spec_file
-
     - name: 'Bump release'
       command: "rpmdev-bumpspec {{ spec_file.files[0].path | basename }} -c \"{{ changelog }}\""
```
Comment (evgeni) [inline obal/data/playbooks/bump-release/bump-release.yaml]:
```suggestion
      command: "rpmdev-bumpspec {{ spec_file_path | basename }} -c \"{{ changelog }}\""
```

**Comment by komidore64** [comment]
@evgeni your suggestions have been incorporated!

**Comment by ekohl** [comment]
> I'd prefer not to do this (yet), as that would require changes to all our downstream packaging repos.

We could start with a configurable warning that could be an error. That would help us make sure foreman-packaging's rpm/develop doesn't regress.

**Comment by komidore64** [comment]
I'm not sure why this blew up.

**Pull request merged**